xamarin/realm-tutorial-dotnet/TaskPage.xaml.cs

```diff
@@ -0,0 +1,135 @@
+    {
+        private Realm taskRealm;
+        private ObservableCollection<Task> _tasks = new ObservableCollection<Task>();
+
+        public ObservableCollection<Task> MyTasks
+        {
+            get
+            {
+                return _tasks;
+            }
+        }
+
+        public TaskPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override async void OnAppearing()
+        {
+            WaitingLayout.IsVisible = true;
+            try
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+                SetUpTaskList();
+            }
+            catch (Exception ex)
+            {
+                await DisplayAlert("Error Fetching Tasks", ex.Message, "OK");
+            }
+            base.OnAppearing();
+        }
+
+        private void SetUpTaskList()
+        {
+            if (_tasks != null || _tasks.Count == 0)
+            {
+                WaitingLayout.IsVisible = true;
+                _tasks = new ObservableCollection<Task>(taskRealm.All<Task>().ToList());
+                listTasks.ItemsSource = _tasks;
+                WaitingLayout.IsVisible = false;
+            }
+        }
+
+        async void New_Button_Clicked(object sender, EventArgs e)
+        {
+            string result = await DisplayPromptAsync("New Task", "Enter the Task Name");
+
+            if (result == null)
+            {
+                return;
+            }
+
+            if (taskRealm == null)
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+            }
+
+            var newTask = new Task()
+            {
+                Summary = result,
+                IsComplete = false
+            };
+
+            taskRealm.Write(() =>
+            {
+                taskRealm.Add(newTask);
+            });
+
+            MyTasks.Add(newTask);
+        }
+
+        void chkCompleted_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            var isCompleteSwitch = (Switch)sender;
+            var changedTask = _tasks.FirstOrDefault(t => t.Id == isCompleteSwitch.AutomationId);
+            if (changedTask != null && e.PropertyName == "IsToggled")
+            {
+                taskRealm.Write(() =>
+                    {
+                        changedTask.IsComplete = isCompleteSwitch.IsToggled;
+                    });
+            }
+        }
+
+        async void Logout_Clicked(object sender, EventArgs e)
+        {
+            if (Navigation.NavigationStack.Count == 1)
+            {
+                await App.RealmApp.CurrentUser.LogOutAsync();
+                var loginPage = new LoginPage();
+                NavigationPage.SetHasBackButton(loginPage, false);
+                await Navigation.PushAsync(loginPage);
+            }
+            else
+            {
+                await Navigation.PopToRootAsync();
+            }
+        }
+
+        async void Delete_Clicked(object sender, EventArgs e)
```

Comment:
This can actually be simplified if you use bindings. In order to do so

```c-sharp
var taskToDelete = (e as TappedEventArgs).Parameter as Task;
```
If you do it like this, then you don't need to "find" the task, because it's automatically passed from the command. This also requires some changes in the xaml, but I'll write them in another comment.